pipeline/merger/src/main/scala/uk/ac/wellcome/platform/merger/services/Merger.scala

```diff
@@ -108,32 +132,53 @@ object PlatformMerger extends Merger {
       case _                                  => None
     }
 
-  override def createMergeResult(target: Work.Visible[Source],
-                                 sources: Seq[Work[Source]]): MergeState =
+  override def createMergeResult(
+    target: Work.Visible[Source],
+    sources: Seq[Work[Source]]): State[MergeState, MergeResult] =
     if (sources.isEmpty)
       State.pure(
         MergeResult(
-          mergedTarget = target.transition[Merged](false),
-          images = ImagesRule.merge(target).data
+          mergedTarget = target.transition[Merged](0),
+          images = standaloneImages(target).map {
+            _ mergeWith (
+              canonicalWork = target.toSourceWork,
+              redirectedWork = None,
+              nMergedSources = 0
+            )
+          }
         )
       )
     else
       for {
-        items <- ItemsRule(target, sources)
-        thumbnail <- ThumbnailRule(target, sources)
-        otherIdentifiers <- OtherIdentifiersRule(target, sources)
-        images <- ImagesRule(target, sources)
+        items <- ItemsRule(target, sources).redirect
+        thumbnail <- ThumbnailRule(target, sources).redirect
+        otherIdentifiers <- OtherIdentifiersRule(target, sources).redirect
+        unmergedImages <- ImagesRule(target, sources).noRedirect
         work = target.mapData { data =>
           data.copy[DataState.Unidentified](
             items = items,
             thumbnail = thumbnail,
             otherIdentifiers = otherIdentifiers,
-            images = images.map(_.toUnmerged)
+            images = unmergedImages
           )
         }
+        nSources <- State.inspect[MergeState, Int](_.size)
       } yield
         MergeResult(
-          mergedTarget = work.transition[Merged](true),
-          images = images
+          mergedTarget = work.transition[Merged](nSources),
+          images = unmergedImages.map { image =>
+            image mergeWith (
+              canonicalWork = work.toSourceWork,
+              redirectedWork = sources
```

Comment:
I made the decision to do this search for the image source because the alternative that I could think of would be to have `ImagesRule` return a kind of intermediary type that might look something like 
```scala
case class ImageMergeResult(image: UnmergedImage, source: Work.Visible[Source])
```
and I decided that the alternative introduced more confusion/complexity/inconsistency compared to the other rules than the search did in this one place.